### PR TITLE
PAAS-2030: Hide secrets in packages and mixins from jelastic console …

### PR DESCRIPTION
**Pull request by amitc-jahia**
…logs in datadog

JIRA issue: https://jira.jahia.org/browse/PAAS-2030

Short description:

### CODE REVIEW

**Review by Ronan-Gorain — CHANGES_REQUESTED** [review]
Other comments : 
mixins/common.yml    
 - Vault token is not hidden line 401                         
 - globals.vaultSecretData is not renamed line 925 and 941     
mixins/jahia.yml    
 - this.__secret__pat_token is directly used line 1339    
    
In onBeforeInstall section of install packages, variables __secret__* should be defined for sensitive informations like datadogapikey, db password and all other kind of sensitive info.
    
From what I remember, there is no need to rename packages paramters. Did you find out it was necessary ? Or was the ticket not clear about it ? If not necessary, I suggest to rollback because this will require to update everything on front side.

File: mixins/common.yml [hunk 1]
```diff
@@ -487,9 +492,14 @@ actions:
         data = JSON.parse(EntityUtils.toString(resp.getEntity()));
         secrets = java.lang.String(data["data"]["data"]);
         secretsB64 = Base64.getEncoder().encodeToString(secrets.getBytes());
-        return {"result": 0, out: secretsB64 };
-    - setGlobals:
-        vaultSecretData: ${response.out}
```
[the remark this answers is not in the document]
Comment (amitc-jahia) [reply mixins/common.yml:492]: Fixed

File: mixins/jcustomer.yml [hunk 2]
```diff
@@ -239,9 +254,10 @@ actions:
         res = ""
         linkedJahiaEnvs.forEach(
           function(jahiaEnvname) {
+            __secret__newPasswordB64 = "${this.new_password_b64}";
```
[the remark this answers is not in the document]
Comment (amitc-jahia) [reply mixins/jcustomer.yml]: Fixed

File: mixins/mariadb.yml [hunk 3]
```diff
@@ -42,8 +42,9 @@ actions:
           user: root
     - log: "## Create DX's Jahia database and user"
     - cmd[${nodes.sqldb.master.id}]: |-
+        __secret__password="${this.__secret__password}"
```
[the remark this answers is not in the document]
Comment (amitc-jahia) [reply mixins/mariadb.yml:45]: Fixed

File: packages/jahia/install.yml [hunk 4]
```diff
@@ -16,7 +16,7 @@ success: ../../assets/jahia/success_install.md
 globals:
   jahia_env_version: 17
   db_user: jahia-db-${fn.random}
-  db_pass: ${fn.password(20)}
```
[the remark this answers is not in the document]
Comment (amitc-jahia) [reply packages/jahia/install.yml:19]: Fixed

**Comment by amitc-jahia** [comment]
Other comments :
    mixins/common.yml
    Vault token is not hidden line 401
    ------> Fixed
    globals.vaultSecretData is not renamed line 925 and 941
    ------> Fixed
    mixins/jahia.yml
    this.__secret__pat_token is directly used line 1339
    ------> Fixed

In onBeforeInstall section of install packages, variables secret* should be defined for sensitive informations like datadogapikey, db password and all other kind of sensitive info.
------> Updated other variables in the install package with sensitive data

From what I remember, there is no need to rename packages paramters. Did you find out it was necessary ? Or was the ticket not clear about it ? If not necessary, I suggest to rollback because this will require to update everything on front side.
------> My bad. Reverted the packages parameters.

**Review by Ronan-Gorain — CHANGES_REQUESTED** [review]
Everything is ok, except the vaultAppRoleLogin action description that was not updated and still says that the action return a global named vaultToken instead of __secret__vaultToken .
sorry, I forgot to mention it in my previous review.

**Comment by amitc-jahia** [comment]
> Everything is ok, except the vaultAppRoleLogin action description that was not updated and still says that the action return a global named vaultToken instead of __secret__vaultToken . sorry, I forgot to mention it in my previous review.

Fixed

**Review by Ronan-Gorain — APPROVED** [review]

**Pull request merged**